x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DestinationIndex.java

```diff
@@ -164,11 +176,12 @@ private static Integer findMaxSettingValue(GetSettingsResponse settingsResponse,
         return properties;
     }
 
-    private static Map<String, Object> createMetadata(String analyticsId, Clock clock) {
+    // Visible for testing
+    static Map<String, Object> createMetadata(String analyticsId, Clock clock, Version version) {
         Map<String, Object> metadata = new HashMap<>();
         metadata.put(CREATION_DATE_MILLIS, clock.millis());
         metadata.put(CREATED_BY, "data-frame-analytics");
-        metadata.put(VERSION, Map.of(CREATED, Version.CURRENT));
+        metadata.put(VERSION, Map.of(CREATED, version.toString()));
```

Comment:
As an aside, the created version is in index settings  (`GET myindex/_settings`) as far as I can tell that happens automatically. 

https://github.com/elastic/elasticsearch/blob/0ec40a0abeb85077285f585263b785a33a66ae39/server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java#L386

Is there a reason to prefer this usage?

Reply:
I think it was originally added not realizing version was already available.

Thinking about it though, I wonder if there is a benefit reading the version from `_meta`. Unlike the index settings, we can overwrite `_meta`. That would leave an open option if in the feature we had to manually upgrade an index and tag it so by updating the version in `_meta`. Does that make sense or am I just sugar coating it? :-)

Reply:
> That would leave an open option if in the feature we had to manually upgrade an index and tag it so by updating the version in `_meta`.

This is what we do with most ML indices.  The `version` in `_meta` refers not to the version the index was created in, but to the version that the mappings were last updated in.  So there's definitely a benefit in us storing a version that's not constrained to be the same as the creation version.  But in this case the field that stores the version is called `created`.  It's really a schema version that we want.

Reply:
This is what `_meta` looks in a dest index created by an analytics job:

```
"_meta" : {
        "analytics" : "test",
        "created_by" : "data-frame-analytics",
        "version" : {
          "created" : "7.9.0"
        },
        "creation_date_in_millis" : 1601374448444
      }
```

Thus, `version` is top level in `_meta` and it can be interpreted as the version of the job that created that index.

Reply:
Right, agreed, the fact that the version is keys as `created` is not ideal. We could change this though it'll come with some BWC code.

Reply:
It is consistent with the other ml indices we should stick to this pattern